core/src/main/java/brooklyn/catalog/internal/CatalogInitialization.java

```diff
@@ -0,0 +1,377 @@
+//        B6) go to C1
+
+        if (initialUri!=null) {
+            populateInitialFromUri(catalog, initialUri, PopulateMode.AUTODETECT);
+            return;
+        }
+        
+        String catalogUrl = managementContext.getConfig().getConfig(BrooklynServerConfig.BROOKLYN_CATALOG_URL);
+        if (Strings.isNonBlank(catalogUrl)) {
+            populateInitialFromUri(catalog, catalogUrl, PopulateMode.AUTODETECT);
+            return;
+        }
+        
+        catalogUrl = Urls.mergePaths(BrooklynServerConfig.getMgmtBaseDir( managementContext.getConfig() ), "catalog.bom");
+        if (new File(catalogUrl).exists()) {
+            populateInitialFromUri(catalog, "file:"+catalogUrl, PopulateMode.YAML);
+            return;
+        }
+        
+        catalogUrl = Urls.mergePaths(BrooklynServerConfig.getMgmtBaseDir( managementContext.getConfig() ), "catalog.xml");
+        if (new File(catalogUrl).exists()) {
+            populateInitialFromUri(catalog, "file:"+catalogUrl, PopulateMode.XML);
+            return;
+        }
+
+        // otherwise look for for classpath:/brooklyn/default.catalog.bom --
+        // there is one on the classpath which says to scan, and provides a few templates;
+        // if one is supplied by user in the conf/ dir that will override the item from the classpath
+        // (TBD - we might want to scan for all such bom's?)
+        
+        catalogUrl = "classpath:/brooklyn/default.catalog.bom";
+        if (new ResourceUtils(this).doesUrlExist(catalogUrl)) {
+            populateInitialFromUri(catalog, catalogUrl, PopulateMode.YAML);
+            return;
+        }
+        
+        log.info("No catalog found on classpath or specified; catalog will not be initialized.");
+        return;
+    }
+    
+    private void populateInitialFromUri(BasicBrooklynCatalog catalog, String catalogUrl, PopulateMode mode) {
+        log.debug("Loading initial catalog from {}", catalogUrl);
+
+        Exception problem = null;
+        Object result = null;
+        
+        String contents = null;
+        try {
+            contents = new ResourceUtils(this).getResourceAsString(catalogUrl);
+        } catch (Exception e) {
+            Exceptions.propagateIfFatal(e);
+            if (problem==null) problem = e;
+        }
+
+        if (contents!=null && (mode==PopulateMode.YAML || mode==PopulateMode.AUTODETECT)) {
+            // try YAML first
+            try {
+                catalog.reset(MutableList.<CatalogItem<?,?>>of());
+                result = catalog.addItems(contents);
+            } catch (Exception e) {
+                Exceptions.propagateIfFatal(e);
+                if (problem==null) problem = e;
+            }
+        }
+        
+        if (result==null && contents!=null && (mode==PopulateMode.XML || mode==PopulateMode.AUTODETECT)) {
+            // then try XML
+            problem = populateInitialFromUriXml(catalog, catalogUrl, problem, contents);
```

Comment:
Minor preference: I think the code would be clearer if the try-catch was in this method (similar to the YAML code above), rather than passing the `problem` in as a parameter in the method call here.